### DISCUSSION TITLE
Löschen von Kunden/Mitgliedern

### DISCUSSION THREAD

**Discussion by budda85** [OpenOlitor/OpenOlitor#140] [Q&A] [answered]
Wovon ist es genau abhängig ob ich ein Mitlgied löschen kann oder nicht?
Bei einigen die ich löschen will, fehlt mir der Löschenknopf in der oberen rechten Ecke.
Das Mitglied hat keine laufenden Abos mehr.
Gruß, Jan

**Comment by AlwinEgger** [comment]
Auch Mitglieder die einmal ein Abo hatten können nicht gelöscht werden. Denn so würden die alten, immer noch auffindbaren Abos die Verbindung verlieren und dies führt potentiell zu Problemen.

Was du allerdings kannst, sind die Mitglieder deaktivieren. Dann erscheinen diese nur noch bei entsprechender Einstellung auf der Übersicht.

Oder geht es dir auch um den Datenschutz? Denn dies hatten wir auch schon diskutiert. Aktuell haben wir hier keine automatisierte Lösung. Sicher kannst du die Datenfelder beim Mitglied löschen resp. die obligatorischen Felder mit einem Standardwert `Gelöscht` befüllen. Sicher nicht sehr schön, klar.

**Comment by budda85** [comment]
Einmal Datenschutz, aber auch um Datenmüll. Jemand der kein Mitglied mehr im Verein ist und auch kein Gemüse und Co bezieht, 
muss ja nicht länger in der Datenbank bleiben.

Also müsste ich vergangene Abos/Anteile bei dem Mitlgied rauslöschen und könnte es dann löschen, richtig?

**Reply by AlwinEgger** [reply, marked as answer]
Nein, du wirst kein Abo löschen können, für welches schon einmal eine Lieferung stattgefunden hat. Denn auch da: Fehlt das Abo, dann fehlt die Verbindung zur Auslieferung.

Es geht hier um die Nachvollziehbarkeit. Die wenigsten ERP-Systeme löschen je Daten. Wenn überhaupt werden diese archiviert.

Ich stimme dir natürlich zu, dass wir bei schützenswerten Daten aufpassen muss und diese nicht prinzipiell für immer vorgehalten werden sollten. Aus dem Grund wäre ein Löschen des Mitglieds wünschenswert bei gleichzeitigem Umhängen der Beziehungen zu einem `Dummy-Gelöschter-Kunde` Eintrag. So würde die notwendige Datenintegrität erhalten.

**Comment by budda85** [comment]
Stellt sich mir gerade noch die Frage wie ich einen Anteil löschen kann. Da ist der Knopf auch ausgeraut.

**Reply by AlwinEgger** [reply]
Genau wie ich oben ausführe: Fehlt das Abo, dann fehlt die Verbindung zur Auslieferung. Ein einmal ausgeliefertes Abo/Anteil kann nicht mehr gelöscht werden.

**Comment by alexander-nbg** [comment]
Ähnliche Überlegungen hatte ich letztens auch für unsere Solawi. Randpunkte:
* Da wir in OpenOlitor vertragliche Informationen halten, müsste eine Aufbewahrungsfrist von mindestens 6 Jahren gelten. Vmtl. eher 10 Jahre.
* Danach verpflichtet uns der Datenschutz zur Löschung der Daten (Löschfrist) da es keine rechtliche Grundlage mehr gibt.
Mein Fazit: eine Automatisierung dafür wäre absolut notwendig, hat aber noch Zeit.

Darüber hinaus verwenden wir OpenOlitor auch für einen Blick in die Historie. Damit wäre das Wenigste für uns Datenmüll.

**Comment by budda85** [comment]
Okay dann weiß ich da erstmal bescheid. Danke
